### PR TITLE
if catching up on buffered jobs fails, fallback to getRepo

### PR DESCRIPTION
**Pull request by whyrusleeping**
im not entirely happy with the flow here, feels kinda jank. The main thing we want to do is just keep trying if the event sync fails instead of returning an error (which results in the users repo getting behind until they send another event).

### CODE REVIEW

**Comment by whyrusleeping** [comment]
this is now a 'clean up FetchAndIndexRepo' PR

File: indexer/indexer.go [hunk 1]
```diff
@@ -412,69 +446,48 @@ func (ix *Indexer) FetchAndIndexRepo(ctx context.Context, job *crawlWork) error
...
+		return err
 	}
-	reposFetched.WithLabelValues("success").Inc()
 
 	// this process will send individual indexing events back to the indexer, doing a 'fast forward' of the users entire history
 	// we probably want alternative ways of doing this for 'very large' or 'very old' repos, but this works for now
 	if err := ix.repomgr.ImportNewRepo(ctx, ai.Uid, ai.Did, bytes.NewReader(repo), &rev); err != nil {
 		span.RecordError(err)
 
 		if ipld.IsNotFound(err) {
```
Comment (ericvolp12) [inline indexer/indexer.go:484]: Do we want to log when this happens? Also can we make the error messages a bit more informative i.e. `failed to full-resync repo after catchup failure: %w` or something?

Comment (whyrusleeping) [reply indexer/indexer.go:484]: yeah, thats a good idea

**Review by ericvolp12 — APPROVED** [review]
lgtm, BGS is prod now for search so pls rollout to staging and test first :P

**Pull request merged**